java-checks/src/main/java/org/sonar/java/checks/RedundantTypeCastCheck.java

```diff
@@ -172,6 +175,24 @@ private static boolean isUnnecessarySubtypeCast(Type childType, TypeCastTree typ
         || isUnnecessaryLambdaCast(childType, parentType));
   }
 
+  private static boolean isAmbiguousMethodReference(TypeCastTree typeCastTree) {
+    MethodReferenceTree methodReferenceTree = (MethodReferenceTree) typeCastTree.expression();
+    Symbol.TypeSymbol methodRefOwner = (Symbol.TypeSymbol) methodReferenceTree.method().symbol().owner();
+    long numberMethodRefsOverloads = getOverloadsCount(methodRefOwner,methodReferenceTree.method().symbol().name());
+
+    MethodInvocationTree methodInvocationTree = (MethodInvocationTree) typeCastTree.parent().parent();
+    Symbol.TypeSymbol owner = (Symbol.TypeSymbol) methodInvocationTree.symbol().owner();
```

Comment:
```
 unknown((Supplier<String>) Overloaded::fff);
```
This code... **does not** raise a `ClassCastException`. :)
But an issue is still reported and I think it makes sense to not report an issue if we don't know the type expected. If the fix is easy, and it makes sense to include it with this PR, we could do it now.
Otherwise, we can simply create a new ticket to track this minor issue.